configserver/src/main/java/com/yahoo/vespa/config/server/session/Session.java

```diff
@@ -134,8 +134,18 @@ public void setAthenzDomain(Optional<AthenzDomain> athenzDomain) {
         sessionZooKeeperClient.writeAthenzDomain(athenzDomain);
     }
 
+    /** Returns application id read from ZooKeeper. Will throw RuntimeException if not found */
     public ApplicationId getApplicationId() { return sessionZooKeeperClient.readApplicationId(); }
 
+    /** Returns application id read from ZooKeeper. Will return Optional.empty() if not found */
+    public Optional<ApplicationId> getOptionalApplicationId() {
+        try {
+            return Optional.of(getApplicationId());
```

Comment:
It would perhaps be better if the `readApplicationId()` returned an optional, and `getApplicationId()` threw on empty? I fear something else may go wrong, and we then believe that was because the id didn't exist, even though it does.

Reply:
Yes, good point, that's better

Reply:
Fixed